extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSource.java

```diff
@@ -116,8 +118,12 @@ private List<S3Object> fetchPrefixedS3Objects() {
         return s3Objects;
     }
 
+    private Collection<S3Object> filterOutFolderFile(List<S3Object> contents) {
+        return contents.stream().filter(object -> !object.key().endsWith("/")).collect(Collectors.toSet());
```

Comment:
why does it transform a `List` to a `Set`?

Reply:
changed in [62ef21a](https://github.com/eclipse-edc/Technology-Aws/pull/417/commits/62ef21a588aef4808200177aade729b622607851)